lib/src/signed_video_h26x_sign.c

```diff
@@ -525,7 +538,7 @@ signed_video_add_nalu_for_signing(signed_video_t *self,
 
       SVI_THROW(generate_sei_nalu(self, &payload, &payload_signature_ptr));
       // Add |payload| to buffer. Will be picked up again when the signature has been generated.
-      add_payload_to_buffer(self, payload, payload_signature_ptr);
+      add_payload_to_buffer(self, payload, payload_signature_ptr, self->last_two_bytes[0]);
```

Comment:
This becomes circular since inside the function you use the first instance to set where the index is